src/actions/channels.ts

```diff
@@ -650,26 +650,28 @@ export function leaveChannel(channelId: string): ActionFunc {
                 team_id: channel.team_id,
                 type: channel.type,
             },
-            meta: {
-                offline: {
-                    effect: () => Client4.removeFromChannel(currentUserId, channelId),
-                    commit: {type: 'do_nothing'}, // redux-offline always needs to dispatch something on commit
-                    rollback: () => {
-                        dispatch(batchActions([
-                            {
-                                type: ChannelTypes.RECEIVED_CHANNEL,
-                                data: channel,
-                            },
-                            {
-                                type: ChannelTypes.RECEIVED_MY_CHANNEL_MEMBER,
-                                data: member,
-                            },
-                        ]));
-                    },
-                },
-            },
         });
 
+        try {
+            await Client4.removeFromChannel(currentUserId, channelId);
+        } catch {
+            dispatch(batchActions([
+                {
+                    type: ChannelTypes.RECEIVED_CHANNEL,
+                    data: channel,
+                },
+                {
+                    type: ChannelTypes.RECEIVED_MY_CHANNEL_MEMBER,
+                    data: member,
+                },
+            ]));
+
+            // The category here may not be the one in which the channel was originally located,
```

Comment:
I spent a bit of time here, but ultimately decided not to worry about recovering the exact category again.

Reply:
Makes sense. Managing the side effects that other operations have on the sidebar is something I really struggled with in that code. That's the reason that I ended up just going back to the server to resync categories so often since it was hard to figure out what the effects of any given operation were. The only other thing I could think to do here would be to get the channel's category before leaving it and then restoring, but then you have to worry about potentially paving over other changes done in the meantime.